app/web/src/store/views.store.ts

```diff
@@ -1572,6 +1610,91 @@ export const useViewsStore = (forceChangeSetId?: ChangeSetId) => {
           });
         },
 
+        // view approval requirement endpoints
+        async CREATE_VIEW_APPROVAL_REQUIREMENT(viewId: ViewId, userId: UserId) {
+          return new ApiRequest({
+            method: "put",
+            url: APPROVAL_REQUIREMENTS_API_PREFIX,
+            params: {
+              entityId: viewId,
+              users: [userId],
+            },
+            onSuccess: () => {
+              this.LIST_VIEW_APPROVAL_REQUIREMENTS(viewId);
+            },
+          });
+        },
+        async REMOVE_VIEW_APPROVAL_REQUIREMENT(
+          definitionId: ApprovalRequirementDefinitionId,
+        ) {
+          return new ApiRequest({
+            method: "delete",
+            url: APPROVAL_REQUIREMENTS_API_PREFIX.concat([definitionId]),
+            onSuccess: () => {
+              const viewId =
+                this.requirementDefinitionsById[definitionId]?.entityId;
+              if (viewId) {
+                this.LIST_VIEW_APPROVAL_REQUIREMENTS(viewId);
+              }
+            },
+          });
+        },
+        async LIST_VIEW_APPROVAL_REQUIREMENTS(viewId: ViewId) {
+          return new ApiRequest({
+            method: "get",
+            url: APPROVAL_REQUIREMENTS_API_PREFIX.concat(["entity", viewId]),
+            onSuccess: (response: ViewApprovalRequirementDefinition[]) => {
+              this.requirementDefinitionsById = {} as Record<
+                ApprovalRequirementDefinitionId,
+                ViewApprovalRequirementDefinition
+              >;
+              for (const def of response) {
+                this.requirementDefinitionsById[def.id] = def;
+              }
+            },
+          });
+        },
+        async ADD_INDIVIDUAL_APPROVER_TO_REQUIREMENT(
+          definitionId: ApprovalRequirementDefinitionId,
+          userId: UserId,
+        ) {
+          return new ApiRequest({
+            method: "put",
+            url: APPROVAL_REQUIREMENTS_API_PREFIX.concat([
+              definitionId,
+              "individual-approver",
+              userId,
+            ]),
+            onSuccess: () => {
+              const viewId =
+                this.requirementDefinitionsById[definitionId]?.entityId;
+              if (viewId) {
+                this.LIST_VIEW_APPROVAL_REQUIREMENTS(viewId);
+              }
+            },
+          });
+        },
+        async REMOVE_INDIVIDUAL_APPROVER_FROM_REQUIREMENT(
+          definitionId: ApprovalRequirementDefinitionId,
+          userId: UserId,
+        ) {
+          return new ApiRequest({
+            method: "delete",
+            url: APPROVAL_REQUIREMENTS_API_PREFIX.concat([
+              definitionId,
+              "individual-approver",
+              userId,
+            ]),
+            onSuccess: () => {
```

Comment:
None of these should be using `onSuccess`, all that data should be coming over the `WsEvent` payload to support multiplayer